lib/raven/integrations/sidekiq.rb

```diff
@@ -5,9 +5,14 @@ module Raven
   class SidekiqCleanupMiddleware
     def call(_worker, _job, _queue)
       yield
+    rescue Exception
+      exception_raised = true
+      raise
     ensure
-      Context.clear!
-      BreadcrumbBuffer.clear!
+      unless exception_raised
```

Comment:
We should have a comment explaining the division of labor between the two middlewares. It's not obvious why this condition is necessary. 

Another other option would be to name the variable something that would make the relation clearer, e.g. naming it `context_cleared_by_error_handler`